macros/generate_model_yml.sql

```diff
@@ -0,0 +1,34 @@
+{% macro generate_model_yml( table_name ) %}
+
+{% set model_yaml=[] %}
+
+{% do model_yaml.append('version: 2') %}
+{% do model_yaml.append('') %}
+{% do model_yaml.append('models:') %}
+{% do model_yaml.append('  - name: ' ~ table_name | lower) %}
+{% do model_yaml.append('    description: ' ~ 'table description here') %}
+{% do model_yaml.append('    columns:') %}
+
+{% set table_relation=api.Relation.create(
```

Comment:
Here you can do:
```suggestion
{% set table_relation=ref(table_name) %}
```

For `generate_source_yaml` we had to use the `api.Relation.create` method, because the dbt DAG doesn't _know_ about the `source` yet.
Fortunately, you've already created the model, so can just use `ref` to get the Relation back.